lib/integer.gi

```diff
@@ -980,25 +992,37 @@ BindGlobal( "IsProbablyPrimeIntWithFail", function( n )
     # make $n$ positive and handle trivial cases
     if n < 0         then n := -n;       fi;
     if n in Primes   then return true;   fi;
+    atomic readonly Primes2 do
     if n in Primes2  then return true;   fi;
-    if n in ProbablePrimes2  then return fail;   fi;
+od;
+atomic readonly ProbablePrimes2 do
+if n in ProbablePrimes2  then return fail;   fi;
+od;
     if n <= 1000     then return false;  fi;
```

Comment:
That's bad code formatting -- but I assume this is the same as in the HPC-GAP lib? Ah well. I also copied some badly formatted C code in PR #239 ;)